### PR TITLE
feat(oas): identify required fields in responses - admin

### PR DESCRIPTION
**Pull request by patrick-medusajs**
## Scope

Admin routes

## What

Add OAS `required` to response schemas.

## Why

Code generator can use the `required` property of a schema to mark fields as optional or not when generating TS types.

## Test

* Run `yarn build`
* Run `yarn openapi:generate`
* Run `yarn redocly preview-docs docs/api/admin/openapi.yaml --config=docs-util/redocly/config.yaml`
* Open the documentation preview URL in a browser (http://127.0.0.1:8080)
* Expect responses to have their fields declared as `required`

### CODE REVIEW

**Review by olivermrbl — APPROVED** [review]
Super nice work @patrick-medusajs!

Curious, why are count, offset, limit included in `required` but not shown in the API reference?

Also, should they be required?

File: packages/medusa/src/api/routes/admin/shipping-options/index.ts [hunk 1]
```diff
@@ -55,6 +55,9 @@ export const defaultRelations = ["region", "profile", "requirements"]
 /**
  * @schema AdminShippingOptionsListRes
  * type: object
+ * required:
+ *   - shipping_options
+ *   - count
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/shipping-options/index.ts:60]: I think we should also have here `offset` and `limit` but they're also missing from the response properties.

**Review by shahednasser — CHANGES_REQUESTED** [review]
I noticed missing required fields for these responses, but I'm not sure if there's a reason behind it:

- `AdminCollectionsDeleteRes`
- `AdminDeleteProductsFromCollectionRes`
- `AdminCustomerGroupsListRes`

File: packages/medusa/src/api/routes/admin/shipping-options/index.ts (hunk 1 above)
Comment (patrick-medusajs) [reply packages/medusa/src/api/routes/admin/shipping-options/index.ts:60]: Good catch 🎣 . Fixed in 5c3615b473681bd2265b434cb935f335c2840436
I also added the missing `required` on the other schemas.

**Review by shahednasser — APPROVED** [review]
LGTM

**Pull request merged**